scripts/bazel_diff.py

```diff
@@ -0,0 +1,170 @@
+#!/usr/bin/env python3
+
+"""Computes the potentially differing rules from some git commit.
+
+Computes the rules that differ between a provided baseline commit and the
+current git commit.
+"""
+
+__copyright__ = """
+Part of the Carbon Language project, under the Apache License v2.0 with LLVM
+Exceptions. See /LICENSE for license information.
+SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+"""
+
+import subprocess
+import argparse
+import tempfile
+import os
+import sys
+
+import scripts_utils
+
+
+def log(s: str) -> None:
+    print(s, file=sys.stderr)
+
+
+def make_bazel_diff_script(bazel: str, tmpdir: str) -> str:
+    bazel_diff_path = os.path.join(tmpdir, "bazel_diff")
+    args = [
+        bazel,
+        "run",
+        f"--script_path={bazel_diff_path}",
+        "//bazel/diff:bazel-diff",
+    ]
+    p = subprocess.run(
+        args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, encoding="utf-8"
+    )
+    if p.returncode != 0:
+        print(p.stderr)
+        exit(f"Bazel run returned {p.returncode}")
+    return bazel_diff_path
+
+
+def compute_hashes(
+    bazel: str, bazel_diff: str, tmpdir: str, prefix: str
+) -> str:
+    hashes_path = os.path.join(tmpdir, f"{prefix}_hashes")
+    args = [
+        bazel_diff,
+        "generate-hashes",
+        f"-b={bazel}",
+        f"-w={os.getcwd()}",
+        hashes_path,
+    ]
+    p = subprocess.run(
+        args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, encoding="utf-8"
+    )
+    if p.returncode != 0:
+        print(p.stderr)
+        exit(f"Bazel diff returned {p.returncode}")
+    return hashes_path
+
+
+def impacted_targets(
+    bazel_diff: str, baseline_hashes: str, current_hashes: str
+) -> str:
+    args = [
+        bazel_diff,
+        "get-impacted-targets",
+        f"-sh={baseline_hashes}",
+        f"-fh={current_hashes}",
+    ]
+    p = subprocess.run(
+        args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, encoding="utf-8"
+    )
+    if p.returncode != 0:
+        print(p.stderr)
+        exit(f"Bazel diff returned {p.returncode}")
+    return p.stdout
+
+
+def filter_targets(bazel: str, targets: str) -> str:
+    args = [
+        bazel,
+        "query",
+        (
+            f"let t = set({targets}) in "
+            f"kind(rule, $t) except attr(tags, manual, $t)"
+        ),
+    ]
+    p = subprocess.run(
+        args, stdout=subprocess.PIPE, stderr=subprocess.PIPE, encoding="utf-8"
+    )
+    if p.returncode != 0:
+        print(p.stderr)
+        exit(f"Bazel run returned {p.returncode}")
+    return p.stdout
+
+
+def git_checkout(commit: str) -> None:
+    subprocess.run(
+        [
+            "git",
+            "checkout",
+            "--quiet",
+            commit,
+        ],
+        check=True,
```

Comment:
Makes sense to me, although now I see you're using subprocess.check_* calls, which way are you trying to switch?

Reply:
Sorry, I was trying to say -- I'll switch to the check_* calls since you suggested them and other scripts are using them.

But to be super clear -- I'm perfectly happy either way, really whatever works best for you (or other Python folks). Just let me know how these should look.

Reply:
Oh, I'm happy to switch to `run` if that's the actual advice. I'd probably not been reading closely. I've been using Python for way too long, so habits aren't always right.

Reply:
Sure, done I think. I'll send a follow-up to update other scripts.